subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/downstream/test/TestDownstreamPlugin.java

```diff
@@ -22,12 +27,30 @@
 
     public void apply(final Project project) {
         IncubatingWarning.warn("downstream-testing plugin");
-        TaskMaker.task(project, TEST_DOWNSTREAM_TASK, TestDownstreamTask.class, new Action<TestDownstreamTask>() {
+
+        project.getPlugins().apply(UploadGistsPlugin.class);
+
+        TestDownstreamTask testDownstreamTask = TaskMaker.task(project, TEST_DOWNSTREAM_TASK, TestDownstreamTask.class, new Action<TestDownstreamTask>() {
             @Override
             public void execute(TestDownstreamTask task) {
                 task.setDescription("Runs all downstream tests.");
             }
         });
+
+        final UploadGistsTask uploadGistsTask = (UploadGistsTask) project.getTasks().findByName(UploadGistsPlugin.UPLOAD_GISTS_TASK);
+
+        uploadGistsTask.setFilesToUpload(
+            project
+                .fileTree(testDownstreamTask.getLogDirectory())
+                .filter(new Spec<File>() {
+                    @Override
+                    public boolean isSatisfiedBy(File file) {
+                        return file.getName().endsWith(".log");
+                    }
+                })
+        );
+
+        testDownstreamTask.finalizedBy(uploadGistsTask);
```

Comment:
Will this work?

testDownstreamTask depends on all TestReleaseTasks. If one of the TestReleaseTasks fails, testDownstreamTask will not be executed and hence it will not be finalized. Also if I run one testReleaseTask explicitly I will not get the gists.

I think that gist task needs to finalize every individual TestReleaseTask.